src/main/java/com/example/SpringExample/service/EmployeeServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.example.SpringExample.service;
+
+import com.example.SpringExample.entity.Employee;
+import com.example.SpringExample.repository.EmployeeRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class EmployeeServiceImpl implements EmployeeService{
+
+    private EmployeeRepository employeeRepository;
+
+    public EmployeeServiceImpl(EmployeeRepository employeeRepository) {
+        this.employeeRepository = employeeRepository;
+    }
+
+    @Override
+    public Employee saveEmployee(Employee employee) {
+        Optional<Employee> savedEmployee = employeeRepository.findByEmail(employee.getEmail());
+        if(savedEmployee.isPresent()){
+            throw new ResourceAlreadyExistsException("Employee already exist with given email:" + employee.getEmail());
+        }
+        return employeeRepository.save(employee);
+    }
+
+    @Override
+    public List<Employee> getAllEmployees() {
+        return employeeRepository.findAll();
+    }
+
+    @Override
+    public Optional<Employee> getEmployeeById(long id) {
+        return employeeRepository.findById(id);
```

Comment:
This method should return an Employee object instead of Optional<Employee>.  Also it should throw Resource not found exception if the resource is not available.